### ISSUE TITLE
Temporal Activities Single/Distributed Mode

### ISSUE THREAD

**Issue by sfmskywalker** [elsa-workflows/elsa-core#1030]
When hosting multiple Elsa nodes, it will index all workflow triggers (start activities that represent a trigger) on each node.

This will cause workflows with e.g. the `Timer`  activity to also execute on each node.
Depending on what the workflow's job is, this may or may not be desirable and it should be up to the workflow author to control what the desired behavior should be.

One way that is currently supported already is to simply set the **Singleton** flag on the workflow definition. This will ensure the workflow will only execute once until it completed, before a new workflow instance gets created.

However, there may be workflows that should start only on one node, but as soon as its started, it may be fine for new instances to be created on the same or other nodes.

For this scenario, we might introduce a setting called e.g. `Multi Node` on the temporal activities `Timer`, `Cron` and `StartAt`. This setting will be `false` by default, which means workflows with a timer will only run on one node by default, but if the user needs the workflow to run on all nodes, then they can enable Multi Node.

**Comment by sfmskywalker** [comment]
I'm cancelling this issue for now - it's a little bit tricky to get right, and one can already control whether or not to run the timer on a single node or all nodes by configuring Quartz clustering.

**Comment by tomy2105** [comment]
By Quartz clustering you mean pointing it to store its data into a database?

Can you please confirm it works with Hangfire also, when Hangfire is configured to store its data in a db? Not just Quartz?

**Comment by sfmskywalker** [comment]
> By Quartz clustering you mean pointing it to store its data into a database?

Yes, and of course you need to tell Quartz to enable clustering with the `UseClustering` call, e.g.:

```csharp
services.AddElsa(elsa => elsa
  .AddQuartzTemporalActivities(configureQuartz: quartz => quartz.UsePersistentStore(store =>
  {
      store.UseJsonSerializer();
      store.UseSqlServer(sqlServerConnectionString);
      store.UseClustering();
})));
```

> Can you please confirm it works with Hangfire also, when Hangfire is configured to store its data in a db? Not just Quartz?

Yes, confirmed. For Hangfire you don't need to do anything special to enable clustering when you configure it to connect to a database, like so:

```csharp
services.AddElsa(elsa => elsa.AddHangfireTemporalActivities(hangfire => hangfire.UseSqlServerStorage(sqlServerConnectionString)));
```

**Comment by tomy2105** [comment]
THanks

**Comment by 12wafu** [comment]
I have Elsa configured to use hangfire and hangfire is backed by postgresql. When I set up a simple timer job it fires many more times that expected. I am running 2 nodes and it is firing around 10 times. Does anyone have any suggestions on how to resolve this issue?

**Comment by sfmskywalker** [comment]
That should definitely not happen under normal circumstances. I tried to run 3 nodes with a workflow containing a timer that ticks every 30 seconds (by default, Hangfire's sweep interval is set to 15 seconds).

Although I see multiple trigger events during startup for some reason I can't explain yet, after that I see only one node in my local cluster generate an event at a given time, at a steady interval of 30 seconds.

In other words, I can't reproduce the behavior you describe. Would it be possible for you to setup a simple project that reproduces this behavior that I can try and run on my machine?
